patches/chromium/disable_compositor_recycling.patch

```diff
@@ -0,0 +1,20 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Andy Locascio <andy@slack-corp.com>
+Date: Wed, 21 Aug 2019 12:09:10 -0700
+Subject: fix: disabling compositor recycling
+
+Compositor recycling is useful for Chrome because there can be many tabs and spinning up a compositor for each one would be costly. In practice, Chrome uses the parent compositor code path of browser_compositor_view_mac.mm; the NSView of each tab is detached when it's hidden and attached when it's shown. For Electron, there is no parent compositor, so we're forced into the "own compositor" code path, which seems to be non-optimal and pretty ruthless in terms of the release of resources. Electron has no real concept of multiple tabs per window, so it should be okay to disable this ruthless recycling altogether in Electron.
+
+diff --git a/content/browser/renderer_host/browser_compositor_view_mac.mm b/content/browser/renderer_host/browser_compositor_view_mac.mm
+index 59e58d693c971742951434f6582140d9179235f2..135e7a384a560f55e5201f108fe1ac2db621fbca 100644
+--- a/content/browser/renderer_host/browser_compositor_view_mac.mm
++++ b/content/browser/renderer_host/browser_compositor_view_mac.mm
+@@ -209,7 +209,7 @@
+ }
+ 
+ void BrowserCompositorMac::SetRenderWidgetHostIsHidden(bool hidden) {
```

Comment:
In what circumstances is this being called with `hidden: true`?

I might say it would be more effective to consider changing `hidden` at the callsite(s) as opposed to unconditionally changing it inside the `SetRenderWidgetHostIsHidden` impl, as that might give us a better opportunity to mitigate risk and understand when this is being set to cause this specific error.

Reply:
hidden is true when the view is marked as occluded.  The goal of this change was to prevent the compositor recycle while still allowing every other thing that handles that observer call to know the window is occluded.

This is why we tested the CPU usage when backgrounded.  We wanted occlusion detection to still work, just this one observer to stop doing it's thing